Core/GDCore/Extensions/Platform.h

```diff
@@ -149,28 +160,18 @@ class GD_CORE_API Platform {
    * \brief Activate or disable the logs on the standard output when
    * loading an extension.
    */
-  void EnableExtensionLoadingLogs(bool enable) { enableExtensionLoadingLogs = enable; };
-
-  /**
-   * \brief Called when the IDE is about to shut down: Take this opportunity for
-   * erasing for example any temporary file.
-   * @deprecated This should be removed.
-   */
-  virtual void OnIDEClosed(){};
-
-  /**
-   * \brief Called when the IDE is initialized and ready to be used.
-   * @deprecated This should be removed.
-   */
-  virtual void OnIDEInitialized(){};
-
-#endif
+  void EnableExtensionLoadingLogs(bool enable) {
+    enableExtensionLoadingLogs = enable;
+  };
 
  private:
   std::vector<std::shared_ptr<PlatformExtension>>
       extensionsLoaded;  ///< Extensions of the platform
   std::map<gd::String, CreateFunPtr>
       creationFunctionTable;  ///< Creation functions for objects
+  std::map<gd::String, InstructionOrExpressionGroupMetadata>
+      instructionOrExpressionGroupMetadata;
+  static InstructionOrExpressionGroupMetadata badInstructionOrExpressionGroupMetadata;

```

Comment:
I'm not sure I understand why this is called `bad...`?

Reply:
This is the null object pattern: https://en.wikipedia.org/wiki/Null_object_pattern
The badXxx objects are objects that are valid objects but empty and returned in case of you trying to access something that is not existing (like here, a group that would not have been declared). 

This is a logical error but we avoid crashing the app by returning a null pointer or something like this (the C++ guarantee that you will get an object, but if you asked for an non existing one, you will get the "null object").